core/node/node_framework/src/task.rs

```diff
@@ -28,20 +28,33 @@
 //! - A task that must be started as soon as possible, e.g. healthcheck server.
 //! - A task that may be a driving force for some precondition to be met.
 
-use std::sync::Arc;
+use std::{
+    fmt::{Display, Formatter},
+    sync::Arc,
+};
 
 use tokio::sync::Barrier;
 
 use crate::service::StopReceiver;
 
+/// A unique human-readable identifier of a task.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct TaskId(pub String);
```

Comment:
I think you may:
- Add `impl From<&str> for TaskId` (will simplify all `id` impls).
- Add `impl Deref for TaskId` with `Target = str` (will simplify usage I guess)
- Add a constructor and make internal representation private to make it a bit more future-proof.

Reply:
Great points, done!